samples/snippets/mf_explicit_model_test.py

```diff
@@ -0,0 +1,163 @@
+        with ml1m_zip.open("ml-1m/movies.dat") as movies_file:
+            movies_content = movies_file.read()
+
+        movies_csv = io.BytesIO(movies_content.replace(b"::", b"@"))
+        movies_config = google.cloud.bigquery.LoadJobConfig()
+        movies_config.source_format = "CSV"
+        movies_config.field_delimiter = "@"
+        movies_config.write_disposition = "WRITE_TRUNCATE"
+        movies_config.schema = [
+            google.cloud.bigquery.SchemaField("movie_id", "INT64"),
+            google.cloud.bigquery.SchemaField("movie_title", "STRING"),
+            google.cloud.bigquery.SchemaField("genre", "STRING"),
+        ]
+        bqclient.load_table_from_file(
+            movies_csv, "bqml_tutorial.movies", job_config=movies_config
+        ).result()
+    # [END bigquery_dataframes_bqml_mf_explicit_upload_movielens]
+
+    # [START bigquery_dataframes_bqml_mf_explicit_create]
+    from bigframes.ml import decomposition
+    import bigframes.pandas as bpd
+
+    # Load data from BigQuery
+    bq_df = bpd.read_gbq(
+        "bqml_tutorial.ratings", columns=("user_id", "item_id", "rating")
+    )
+
+    # Create the Matrix Factorization model
+    model = decomposition.MatrixFactorization(
+        num_factors=34,
+        feedback_type="explicit",
+        user_col="user_id",
+        item_col="item_id",
+        rating_col="rating",
+        l2_reg=9.83,
+    )
+    model.fit(bq_df)
+    model.to_gbq(
+        your_model_id, replace=True  # For example: "bqml_tutorial.mf_explicit"
+    )
+    # [END bigquery_dataframes_bqml_mf_explicit_create]
+    # [START bigquery_dataframes_bqml_mf_explicit_evaluate]
+    # Evaluate the model using the score() function
+    model.score()
+    # Output:
+    # mean_absolute_error	mean_squared_error	mean_squared_log_error	median_absolute_error	r2_score	explained_variance
+    # 0.485403	                0.395052	        0.025515	            0.390573	        0.68343	        0.68343
+    # [END bigquery_dataframes_bqml_mf_explicit_evaluate]
+    # [START bigquery_dataframes_bqml_mf_explicit_recommend_df]
+    # Use predict() to get the predicted rating for each movie for 5 users
+    subset = bq_df.head(5)
+    predicted = model.predict(subset)
+    print(predicted)
+    # Output:
+    #   predicted_rating	user_id	 item_id	rating
+    # 0	    4.206146	     4354	  968	     4.0
+    # 1	    4.853099	     3622	  3521	     5.0
+    # 2	    2.679067	     5543	  920	     2.0
+    # 3	    4.323458	     445	  3175	     5.0
+    # 4	    3.476911	     5535	  235	     4.0
+    # [END bigquery_dataframes_bqml_mf_explicit_recommend_df]
+    # [START bigquery_dataframes_bqml_mf_explicit_recommend_model]
+    # import bigframes.bigquery as bbq
+
+    # Load movies
+    movies = bpd.read_gbq("bqml_tutorial.movies")
+
+    # Merge the movies df with the previously created predicted df
+    merged_df = bpd.merge(predicted, movies, left_on="item_id", right_on="movie_id")
```

Comment:
`predicted` here is only a subset. The SQL shows creating a table of all the predictions.

```
CREATE OR REPLACE TABLE `bqml_tutorial.recommend`
AS
SELECT
  *
FROM
  ML.RECOMMEND(MODEL `bqml_tutorial.mf_explicit`);
```

Did we implement a way for `predict()` to work without any inputs like in the SQL?

```suggestion
    predicted = model.predict()

    # Load movies
    movies = bpd.read_gbq("bqml_tutorial.movies")

    # Merge the movies df with the previously created predicted df
    merged_df = bpd.merge(predicted, movies, left_on="item_id", right_on="movie_id")
```

Reply:
This feature isn't supported yet and an issue will be filed shortly

Reply:
currently not supported, issue will be filed